pkg/inject/inject.go

```diff
@@ -351,7 +340,11 @@ func (conf *ResourceConfig) parse(bytes []byte) error {
 
 		conf.workload.obj = v
 		conf.pod.spec = &v.Spec
+		ns := conf.pod.Meta.Namespace
 		conf.pod.Meta = &v.ObjectMeta
+		if ns != "" {
+			conf.pod.Meta.Namespace = ns
```

Comment:
The namespace is missing in the `AdmissionRequest` metadata, so the idea here is to fill it in. It occurred to me that was a natural place to put it (already supported in the `Meta` struct), instead of augmenting `ResourceConfig`.
@ihcsim made me realize we're not using namespace anywhere in the injection logic for pods (which is the webhook case, which is the case where the namespace has to be "filled-in") so we can get rid of all this.

_However_:

We're using the namespace when injecting Deployments through the CLI [here](https://github.com/linkerd/linkerd2/blob/alpeb/webhook-pods/pkg/inject/inject.go#L272-L277) to be able to set `destinationDNSOverride`. 
So we could actually be needing to use namespaces for this same purpose in the particular case of having the webhook inject a linkerd controller pod that wasn't injected to begin with. We're currently not supporting that case, although it would be a very rare and non-default one. Do you think it's worth to consider it?

Reply:
Ahhh, ok great, thanks for the additional explanation. That all makes sense. I'm in favor of reverting these changes, and not worrying about the case where we might auto-inject the proxy into the control plane. We should assume we're always generating injected YAML as part of the install path.